.github/workflows/tests.yml

```diff
@@ -24,20 +24,21 @@ jobs:
           python-version: "3.11"
           architecture: x64
           cache: pip
-          cache-dependency-path: "**/*requirements*.txt"
+          cache-dependency-path: "**/pyproject.toml"
+      - name: Install Poetry
+        run: |
+          pip install poetry
       - name: Install dependencies
         run: |
-          pip install -r code/requirements.txt -r code/dev-requirements.txt -r code/backend/requirements.txt
+          poetry install
       - name: Run Unit tests
-        working-directory: ./code
-        run: python -m pytest -m "not azure and not functional" --junitxml=coverage-junit.xml --cov=. --cov-report xml:coverage.xml
+        run: poetry run pytest -m "not azure and not functional" --junitxml=coverage-junit.xml --cov=. --cov-report xml:coverage.xml
```

Comment:
We'll need to move the `.coveragerc` file into the root of the repo, and update the path inside the file for the tests directory